src/app/my-messages/my-messages.component.ts

```diff
@@ -0,0 +1,101 @@
+import { ChangeDetectorRef, Component, OnInit } from '@angular/core';
+import { HttpErrorResponse } from '@angular/common/http';
+import { of } from 'rxjs';
+import { CommunityRegistrationService } from '../shared/community-registration.service';
+import { CommunityUserRegistration } from '../shared/community-user-registration';
+import { MessagesService } from './messages.service';
+import { DeleteConfirmationDialogComponent } from '../shared/delete-confirmation-dialog/delete-confirmation-dialog.component';
+import { MatDialog } from '@angular/material';
+import { AbstractNotification } from './abstract-notification';
+import { CommunityInvitationNotification } from './community-invitation-notification';
+import { JoinCommunityRequestNotification } from './join-community-request-notification';
+import { GlobalErrorHandlerService } from '../error/global-error-handler.service';
+
+@Component({
+  selector: 'app-my-messages',
+  templateUrl: './my-messages.component.html',
+  styleUrls: ['./my-messages.component.scss']
+})
+export class MyMessagesComponent implements OnInit {
+
+  errorMessage: string = null;
+  notifications$ = of([]);
+
+  constructor(private communityRegistrationService: CommunityRegistrationService,
+              private messageService: MessagesService,
+              public dialog: MatDialog,
+              private changeDetector: ChangeDetectorRef,
+              private globalErrorHandlerService: GlobalErrorHandlerService) {
+  }
+
+  ngOnInit() {
+   this.notifications$ = this.messageService.getUserNotifications();
+  }
+
+  hasJoinRequestType<T extends AbstractNotification>(notification: T): boolean {
+    return notification.hasCommunityInvitationType() || notification.hasJoinCommunityType();
+  }
+
+  showAcceptDeclineButtons<T extends AbstractNotification>(notification: T) {
+    if (notification.hasCommunityInvitationType()) {
```

Comment:
Can we use `Util.createNotificationInstance` here instead of explicit casting?

Reply:
thank you. done